packages/flutter/lib/src/material/material.dart

```diff
@@ -320,7 +320,10 @@ class _MaterialState extends State<Material> with TickerProviderStateMixin {
   @override
   Widget build(BuildContext context) {
     final Color backgroundColor = _getBackgroundColor(context);
-    assert(backgroundColor != null || widget.type == MaterialType.transparency);
+    assert(
+      backgroundColor != null || widget.type == MaterialType.transparency,
+      'If Material type is not MaterialType.transparency, the color property must not be null'
```

Comment:
This appears to be a bit more complicated. This assertion fails if `color` isn't specified and  the `type` isn't MaterialType.transparency _and_ the theme's canvasColor is null (if type is canvas) or the theme's cardColor is null (if the type is card).  Which seems rather unlikely. How did this failure mode come up?

Reply:
It is a little vague in the [issue](https://github.com/flutter/flutter/issues/12321) what the original user's situation was, since they only posted the error message. As far as I can tell, they tried to create their own `Material` with type `MaterialType.button` but did not pass in a `color`.

Reply:
I'll make an update to add more clarification